### PR TITLE
Split Microsoft.EntityFrameworkCore.Analyzers into language-specific analyzers

### LINKED ISSUE #22177: Test runner failing on release/5.0 because of new VB analyzer dependency
Main branch is working fine.

Error:
```
CSC : error AD0001: Analyzer 'Microsoft.EntityFrameworkCore.InternalUsageDiagnosticAnalyzer' threw an exception of type
'System.IO.FileNotFoundException' with message 
'Could not load file or assembly 'Microsoft.CodeAnalysis.VisualBasic, Version=3.7.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.
[/home/ajcvickers/efcore/src/EFCore.Proxies/EFCore.Proxies.csproj]
```
for all the tests.

### PR DESCRIPTION
**Pull request by sharwell**
Fixes #22177

### CODE REVIEW

**Comment by roji** [comment]
Oh wow, thanks for submitting this! We'll have a chat in EF triage a bit later today about it.

File: src/EFCore.Analyzers.CSharp/AnalyzerReleases.Shipped.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+## Release 2.1.0
+
+### New Rules
```
Comment (Youssef1313) [inline src/EFCore.Analyzers.CSharp/AnalyzerReleases.Shipped.md:3]:
```suggestion
### New Rules

```

File: src/EFCore.Analyzers.CSharp/AnalyzerReleases.Shipped.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+Rule ID | Category | Severity | Notes
+--------|----------|----------|-------
+EF1000  | Usage    | Warning  | RawSqlStringInjectionDiagnosticAnalyzer, [Documentation](https://docs.microsoft.com/ef/core/querying/raw-sql)
+
+## Release 3.0.0
+
+### New Rules
```
Comment (Youssef1313) [inline src/EFCore.Analyzers.CSharp/AnalyzerReleases.Shipped.md:10]:
```suggestion
### New Rules

```

File: src/EFCore.Analyzers.CSharp/AnalyzerReleases.Shipped.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+Rule ID | Category | Severity | Notes
+--------|----------|----------|-------
+
+### Removed Rules
```
Comment (Youssef1313) [inline src/EFCore.Analyzers.CSharp/AnalyzerReleases.Shipped.md:14]:
```suggestion
### Removed Rules

```

**Review by Youssef1313 — CHANGES_REQUESTED** [review]
nit: some markdown parsers require blank lines around tables.
Also, [MD022](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md022---headings-should-be-surrounded-by-blank-lines) requires blank lines around headings.

File: src/EFCore.Analyzers.VisualBasic/AnalyzerReleases.Shipped.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+## Release 2.1.0
+
+### New Rules
```
Comment (Youssef1313) [inline src/EFCore.Analyzers.VisualBasic/AnalyzerReleases.Shipped.md:3]:
```suggestion
### New Rules

```

File: src/EFCore.Analyzers.VisualBasic/AnalyzerReleases.Shipped.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+Rule ID | Category | Severity | Notes
+--------|----------|----------|-------
+EF1000  | Usage    | Warning  | RawSqlStringInjectionDiagnosticAnalyzer, [Documentation](https://docs.microsoft.com/ef/core/querying/raw-sql)
+
+## Release 3.0.0
+
+### New Rules
```
Comment (Youssef1313) [inline src/EFCore.Analyzers.VisualBasic/AnalyzerReleases.Shipped.md:10]:
```suggestion
### New Rules

```

File: src/EFCore.Analyzers.VisualBasic/AnalyzerReleases.Shipped.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+Rule ID | Category | Severity | Notes
+--------|----------|----------|-------
+
+### Removed Rules
```
Comment (Youssef1313) [inline src/EFCore.Analyzers.VisualBasic/AnalyzerReleases.Shipped.md:14]:
```suggestion
### Removed Rules

```

File: src/EFCore.Analyzers.VisualBasic/VisualBasicInternalUsageDiagnosticAnalyzer.cs [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+{
+    [DiagnosticAnalyzer(LanguageNames.VisualBasic)]
+    public class VisualBasicInternalUsageDiagnosticAnalyzer : AbstractInternalUsageDiagnosticAnalyzer
+    {
+        protected override SyntaxNodeOrToken GetLocationForDiagnostic(SyntaxNode syntax)
+        {
+            return syntax switch
+            {
+                _ => syntax,
+            };
```
Comment (Youssef1313) [inline src/EFCore.Analyzers.VisualBasic/VisualBasicInternalUsageDiagnosticAnalyzer.cs:18]: Why not return syntax directly?

**Review by Youssef1313 — CHANGES_REQUESTED** [review]

File: src/EFCore.Analyzers.CSharp/AnalyzerReleases.Shipped.md (hunk 1 above)
Comment (sharwell) [reply src/EFCore.Analyzers.CSharp/AnalyzerReleases.Shipped.md:3]: These are direct copied from the existing files in the repository.

File: src/EFCore.Analyzers.CSharp/AnalyzerReleases.Shipped.md (hunk 2 above)
Comment (sharwell) [reply src/EFCore.Analyzers.CSharp/AnalyzerReleases.Shipped.md:10]: ➡️ See https://github.com/dotnet/efcore/pull/22195#discussion_r475795538

File: src/EFCore.Analyzers.VisualBasic/AnalyzerReleases.Shipped.md (hunk 6 above)
Comment (sharwell) [reply src/EFCore.Analyzers.VisualBasic/AnalyzerReleases.Shipped.md:14]: ➡️ See https://github.com/dotnet/efcore/pull/22195#discussion_r475795538

File: src/EFCore.Analyzers.VisualBasic/AnalyzerReleases.Shipped.md (hunk 5 above)
Comment (sharwell) [reply src/EFCore.Analyzers.VisualBasic/AnalyzerReleases.Shipped.md:10]: ➡️ See https://github.com/dotnet/efcore/pull/22195#discussion_r475795538

File: src/EFCore.Analyzers.VisualBasic/AnalyzerReleases.Shipped.md (hunk 4 above)
Comment (sharwell) [reply src/EFCore.Analyzers.VisualBasic/AnalyzerReleases.Shipped.md:3]: ➡️ See https://github.com/dotnet/efcore/pull/22195#discussion_r475795538

File: src/EFCore.Analyzers.CSharp/AnalyzerReleases.Shipped.md (hunk 3 above)
Comment (sharwell) [reply src/EFCore.Analyzers.CSharp/AnalyzerReleases.Shipped.md:14]: ➡️ See https://github.com/dotnet/efcore/pull/22195#discussion_r475795538

File: src/EFCore.Analyzers.VisualBasic/VisualBasicInternalUsageDiagnosticAnalyzer.cs (hunk 7 above)
Comment (sharwell) [reply src/EFCore.Analyzers.VisualBasic/VisualBasicInternalUsageDiagnosticAnalyzer.cs:18]: It seems likely that someone would want to alter this in the future to provide more specific results for VB.

**Comment by roji** [comment]
Postponing merging this PR as per https://github.com/dotnet/efcore/issues/22177#issuecomment-679304818

**Comment by smitpatel** [comment]
This can be rebased and merged to main now.

**Comment by roji** [comment]
Note: am going to defer working on this until we make a decision on what happens with pubternality... Even though we may still want to provide analyzers for other purposes, this is the only one we currently have.

**Comment by AraHaan** [comment]
Is there plans to continue this soon?

**Comment by dougbu** [comment]
If this PR or something like it is needed still, please retarget release/6.0

**Pull request closed without merging**